core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala

```diff
@@ -70,17 +70,22 @@ private[kafka] abstract class SingleSourceLogic[K, V, Msg](
       // Is info too much here? Will be logged at startup / rebalance
       tps ++= newTps
       log.log(partitionLogLevel, "Assigned partitions: {}. All partitions: {}", newTps, tps)
+      notifyUserOnAssign(newTps)
       requestMessages()
     }
 
     val partitionRevokedCB = getAsyncCallback[Set[TopicPartition]] { newTps =>
       tps --= newTps
       log.log(partitionLogLevel, "Revoked partitions: {}. All partitions: {}", newTps, tps)
-      requestMessages()
+      notifyUserOnRevoke(newTps)
+      finally requestMessages()
```

Comment:
finally not needed any more (and where is the `try`?

Reply:
yep